File: localization/src/main/java/com/prongbang/localization/LocalizationApplication.java
package com.prongbang.localization;

import android.app.Application;
import android.content.Context;

import com.prongbang.localization.utils.ConstantUtil;
import com.prongbang.localization.utils.LocaleHelper;

/**
 * Created by prongbang on 11/22/2017 AD.
 */

public class LocalizationApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        LocaleHelper.setLocale(this);
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(LocaleHelper.onAttach(base, ConstantUtil.LOCALE_DEFAULT));
    }

}

File: localization/src/main/java/com/prongbang/localization/LocalizationApplication.kt
package com.prongbang.localization

import android.app.Application
import android.content.Context
import java.util.*

open class LocalizationApplication : Application(), LocalizationManager {

	override fun onCreate() {
		super.onCreate()
		LocalizeManager.initLocale(this)
	}

	override fun attachBaseContext(base: Context) {
		super.attachBaseContext(LocalizeManager.onAttach(base))
	}

	override fun setLocale(locale: Locale) {
		LocalizeManager.changeLocale(this, locale)
	}
}